### PR TITLE
Postgres search

### PR DESCRIPTION
**Pull request by zaziemo**
After trying out to build the search from scratch and using "simple" postgres features we decided to use pg_search for the core search functionality. 
Due to performance requirements @A5308Y built the `ProfileGrouper` to  provide the data for our facets to filter the search results. 
Besides we replaced kaminari with pagy for pagination to further increase the display of the search results.
We deleted all the elasticsearch related code and rewrote some deprecated controller specs as system specs.

Solves the issues #1316 #1269

### CODE REVIEW

File: app/controllers/admin/profiles_controller.rb [hunk 1]
```diff
@@ -6,11 +6,25 @@ class Admin::ProfilesController < Admin::BaseController
   before_action :set_profile, only: %i[show edit update destroy publish unpublish admin_update]
 
   def index
-    @profiles = if params[:search]
-                  Profile.is_confirmed.admin_search(params[:search]).order('profiles.created_at DESC').page(params[:page]).per(100)
-                else
-                  Profile.is_confirmed.order(sort_column + ' ' + sort_direction).order('created_at DESC').page(params[:page]).per(100)
-                end
+    @paginated_profiles =
```
Comment (zaziemo) [inline app/controllers/admin/profiles_controller.rb:9]: Why do I use the variable name? Where do we need it?

File: app/controllers/pages_controller.rb [hunk 2]
```diff
@@ -29,6 +34,46 @@ def render_footer?
...
+      results_array << {
+                      title: feature.title,
+                      description: feature.description,
+                      profiles: profiles_array(feature.profiles)
+                    }
+    end
+    results_array
+  end
+
+  def profiles_array(profiles)
```
Comment (tyranja) [inline app/controllers/pages_controller.rb:51]: Could this be refactored to?

```
def profiles_array(profiles)
  profiles.map do |profile|
    {
      id: profile.id,
      fullname: profile.fullname,
      iso_languages: profile.iso_languages,
      city: profile.city,
      willing_to_travel: profile.willing_to_travel,
      nonprofit: profile.nonprofit,
      main_topic_or_first_topic: profile.main_topic_or_first_topic
    }
  end
end
```

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -29,6 +34,46 @@ def render_footer?
 
   private
 
+  def features_and_profiles
```
Comment (tyranja) [inline app/controllers/pages_controller.rb:37]: I think we could write:

```
def features_and_profiles
  feature_records = Feature.published_feature.order(:position)

  feature_records.map do |feature|
    {
      title: feature.title,
      description: feature.description,
      profiles: profiles_array(feature.profiles)
    }
  end
end
```

File: app/controllers/profiles_controller.rb [hunk 4]
```diff
@@ -131,36 +132,45 @@ def profile_params
...
+    @aggs_languages = aggs[:languages]
+    @aggs_cities = aggs[:cities]
+    @aggs_states = aggs[:states]
+    @aggs_countries = aggs[:countries]
   end
 
-  def profiles_for_search
-    Profile
+  def matching_profiles
+    chain ||= Profile
```
Comment (tyranja) [inline app/controllers/profiles_controller.rb:146]: Maybe call it `query_results`

File: app/models/profile.rb [hunk 5]
```diff
@@ -93,6 +116,18 @@ def self.typeahead(term, region: nil)
     suggestions.map { |s| s.downcase }.uniq
   end
 
+  def to_card_hash
```
Comment (tyranja) [inline app/models/profile.rb:119]: `profile_view_details` ?

File: app/models/profile.rb [hunk 6]
```diff
@@ -112,7 +147,7 @@ def name_or_email
   end
 
   def main_topic_or_first_topic
-    main_topic.present? ? main_topic : topic_list.first
+    main_topic.present? ? main_topic : taggings.map { |tagging| tagging.tag.name }.first
```
Comment (tyranja) [inline app/models/profile.rb:150]: why did you do that?

File: app/views/profiles/_profile.html.erb [hunk 7]
```diff
@@ -1,15 +1,15 @@
 <% if profile %>
-  <% tooltip_empty = !profile.iso_languages.present? && profile.city.nil? && profile.willing_to_travel.nil? && profile.nonprofit.nil? %>
+  <% tooltip_empty = !profile[:iso_languages].present? && profile[:city].nil? && profile[:willing_to_travel].nil? && profile[:nonprofit].nil? %>
   <% title = render partial: "profiles/profile_tooltip", locals: { profile: profile } %>
   <div class="col-sm-6 col-md-6 col-lg-3 my-3 profile-box">
-    <%= profile_image_link(profile) %>
+    <%= profile_image_link(Profile.find(profile[:id])) %>
```
Comment (tyranja) [inline app/views/profiles/_profile.html.erb:5]: Why not using profile? Here you have to grep it again from the DB

**Review by tyranja — APPROVED** [review]
Thanks for all the work. I have added some suggestions. But nothing urgent ...

File: app/controllers/pages_controller.rb (hunk 2 above)
Comment (zaziemo) [reply app/controllers/pages_controller.rb:51]: Good idea to refactor. I did it with the new method in profile.rb `feature.profiles.map(&:to_card_hash)` and removed the `profiles_array` method altogether.

File: app/controllers/pages_controller.rb (hunk 3 above)
Comment (zaziemo) [reply app/controllers/pages_controller.rb:37]: done :)

File: app/models/profile.rb (hunk 5 above)
Comment (zaziemo) [reply app/models/profile.rb:119]: I decided for `profile_view_details`

File: app/models/profile.rb (hunk 6 above)
Comment (zaziemo) [reply app/models/profile.rb:150]: The `topic_list` method created a n+1 query as fas as I remeber.

File: app/views/profiles/_profile.html.erb (hunk 7 above)
Comment (zaziemo) [reply app/views/profiles/_profile.html.erb:5]: In this case `profile` is not the object itself but the hash with the limited attributes. But the `profile_image_link` method needs the profile instance, doesn't it?

File: app/controllers/profiles_controller.rb (hunk 4 above)
Comment (zaziemo) [reply app/controllers/profiles_controller.rb:146]: do you mean the method or `chain`?

**Pull request merged**